sqlalchemy_redshift/dialect.py

```diff
@@ -625,6 +620,84 @@ def get_columns(self, connection, table_name, schema=None, **kw):
             columns.append(column_info)
         return columns
 
+    @reflection.cache
+    def has_table(self, connection, table_name, schema=None, **kw):
+        if not schema:
+            schema = inspect(connection).default_schema_name
+
+        info_cache = kw.get('info_cache')
+        table = self._get_all_relation_info(connection,
+                                            schema=schema,
+                                            table_name=table_name,
+                                            info_cache=info_cache)
+
+        return True if table else False
+
+    @reflection.cache
+    def get_check_constraints(self, connection, table_name, schema=None, **kw):
+        table_oid = self.get_table_oid(
+            connection, table_name, schema, info_cache=kw.get("info_cache")
+        )
+        table_oid = 'NULL' if not table_oid else table_oid
+
+        result = connection.execute("""
+                        SELECT
+                            cons.conname as name,
+                            pg_get_constraintdef(cons.oid) as src
+                        FROM
+                            pg_catalog.pg_constraint cons
+                        WHERE
+                            cons.conrelid = {} AND
+                            cons.contype = 'c'
+                        """.format(table_oid))
+        ret = []
+        for name, src in result:
+            # samples:
+            # "CHECK (((a > 1) AND (a < 5)))"
+            # "CHECK (((a = 1) OR ((a > 2) AND (a < 5))))"
+            # "CHECK (((a > 1) AND (a < 5))) NOT VALID"
+            # "CHECK (some_boolean_function(a))"
+            # "CHECK (((a\n < 1)\n OR\n (a\n >= 5))\n)"
+
+            m = re.match(
+                r"^CHECK *\((.+)\)( NOT VALID)?$", src, flags=re.DOTALL
+            )
+            if not m:
+                print(f"Could not parse CHECK constraint text: {src}")
```

Comment:
do we want to print or log here? idk if we have existing log statements in the project, but it could be good to introduce one in this case

Reply:
Logging would be better, I can change this.